app/Gists/File.php

```diff
@@ -0,0 +1,64 @@
+<?php namespace Gistlog\Gists;
+
+use Illuminate\Contracts\Support\Arrayable;
+
+class File implements Arrayable
+{
+    public $name;
+    public $type;
+    public $language;
+    public $url;
+    public $size;
+    public $content;
+
+    /**
+     * @param array|ArrayAccess $rawFile
+     * @return File
+     */
+    public static function fromGitHub($rawFile)
+    {
+        $file = new self;
+
+        $file->name = $rawFile['filename'];
+        $file->type = $rawFile['type'];
+        $file->language = $rawFile['language'];
+        $file->url = $rawFile['raw_url'];
+        $file->size = $rawFile['size'];
+        $file->content = $rawFile['content'];
+
+        return $file;
+    }
+
+    /**
+     * @param array|ArrayAccess $rawFiles
+     * @return Collection
+     */
+    public static function multipleFromGitHub($rawFiles)
+    {
+        // Remove the blog post itself.
+        array_shift($rawFiles);
```

Comment:
Thanks for taking a look @mattstauffer! At present when [importing from Github][get from GitHub] we assume that the first file is the blog post. 

We could have a `FilesCollection` that takes all the raw files and has some additional methods such as one to best guess the users intended `blog.md` file following this kind of flow: 

- is there a file specified in the `gistlog.yml`. 
- is there a file called `blog.md`. Maybe have this as a default in the `gistlog.yml`.
- fallback to the first file in the collection.

This same collection could be quite useful to extract the other files to be used in my PR so I'd be happy to implement this.

Now as you have suggested @JacobBennett, we can then encourage people to use `blog.md`, support people who want to call is `my_awesome_post.md` while still supporting break current Gistlogs.

[get from GitHub]: https://github.com/tightenco/gistlog/blob/master/app/Gists/Gistlog.php#L45-L46